game-core/src/main/java/games/strategy/triplea/attachments/AbstractPlayerRulesAttachment.java

```diff
@@ -78,6 +80,48 @@ public static RulesAttachment get(final PlayerID player) {
     return rulesAttachment;
   }
 
+  /**
+   * Get condition attachment for the given player and condition name.
+   */
+  public static ICondition getCondition(final String playerName, final String conditionName,
+      final GameData data) {
+    final PlayerID player = data.getPlayerList().getPlayerId(playerName);
+    if (player == null) {
+      // could be an old map, or an old save, so we don't want to stop the game from running.
+      System.err.println("When trying to find condition: " + conditionName + ", player does not exist: " + playerName);
+      return null;
+    }
+    final Collection<PlayerID> allPlayers = data.getPlayerList().getPlayers();
+    final IAttachment attachment;
+    try {
+      if (conditionName.contains(Constants.RULES_OBJECTIVE_PREFIX)
+          || conditionName.contains(Constants.RULES_CONDITION_PREFIX)) {
+        attachment = RulesAttachment.get(player, conditionName, allPlayers, true);
+      } else if (conditionName.contains(Constants.TRIGGER_ATTACHMENT_PREFIX)) {
+        attachment = TriggerAttachment.get(player, conditionName, allPlayers);
+      } else if (conditionName.contains(Constants.POLITICALACTION_ATTACHMENT_PREFIX)) {
+        attachment = PoliticalActionAttachment.get(player, conditionName, allPlayers);
+      } else {
+        System.err.println(conditionName + " attachment must begin with: " + Constants.RULES_OBJECTIVE_PREFIX
+            + " or " + Constants.RULES_CONDITION_PREFIX + " or " + Constants.TRIGGER_ATTACHMENT_PREFIX + " or "
+            + Constants.POLITICALACTION_ATTACHMENT_PREFIX);
+        return null;
+      }
+    } catch (final Exception e) {
+      // could be an old map, or an old save, so we don't want to stop the game from running.
+      System.err.println(e.getMessage());
+      return null;
+    }
+    if (attachment == null) {
+      System.err.println("Condition attachment does not exist: " + conditionName);
+      return null;
+    }
+    if (!ICondition.class.isAssignableFrom(attachment.getClass())) {
```

Comment:
Can't this be simplified to a simple instanceof check?

Reply:
Yeah, though I don't see any way for that to ever occur so I'm just gonna remove that check entirely.